File: Administration/DataBase/MissingPerson.js
var mongoose = require('mongoose');

option = { useNewUrlParser: true };
mongoose.connect("mongodb+srv://zweichen:G2ITBTCES@cluster0-75ahh.mongodb.net/test?retryWrites=true").then(
  () => { console.log('Successfully connected to MongoDB!'); },
  err => { console.log('Error when connecting to MongoDB! ' + err); }
);

var Schema = mongoose.Schema;

var missingPersonSchema = new Schema({
    name: {type: String, required: true, unique: true},
    lastSeenStreet: {type: String, required: true},
    lastSeenCity: {type: String, required: true},
    lastSeenState: {type: String, required: true},
    description: {type: String, required: true},
    contactNumber: {type: Number, required: true},
    isRequest: {type: Boolean, required: true},
    image: {type: String, required: false},
    imageBase64: {type: String, required: false},
    hasImage: {type: Boolean, required: true}
});

module.exports = mongoose.model('MissingPerson', missingPersonSchema);

File: ConnectSentinel/app/src/main/java/group2/connectsentinel/data/MissingPerson.java
package group2.connectsentinel.data;

public class MissingPerson {

    private String lastSeenStreet;
    private String lastSeenCity;
    private String lastSeenState;
    private String description;
    private String name;
    private long contact;
    private String image;
    private boolean hasImage;

    public MissingPerson(String lastSeenStreet, String lastSeenCity, String lastSeenState,
                         String description, String name, long contact, boolean hasImage) {
        this.lastSeenStreet = lastSeenStreet;
        this.lastSeenCity = lastSeenCity;
        this.lastSeenState = lastSeenState;
        this.description = description;
        this.name = name;
        this.contact = contact;
        this.hasImage = hasImage;
    }

    public MissingPerson(String lastSeenStreet, String lastSeenCity, String lastSeenState,
                         String description, String name, long contact, String image,
                         boolean hasImage) {
        this.lastSeenStreet = lastSeenStreet;
        this.lastSeenCity = lastSeenCity;
        this.lastSeenState = lastSeenState;
        this.description = description;
        this.name = name;
        this.contact = contact;
        this.image = image;
        this.hasImage = hasImage;
    }

    public String getDescription() {
        return description;
    }

    public String getLastSeenStreet() {
        return lastSeenStreet;
    }

    public String getLastSeenCity() {
        return lastSeenCity;
    }

    public String getLastSeenState() {
        return lastSeenState;
    }

    public String getName() {
        return name;
    }

    public long getContact() {
        return contact;
    }

    public String getImage() {
        return image;
    }

    public boolean getHasImage() {
        return hasImage;
    }
}
